cmake/modules/kconfig.cmake

```diff
@@ -29,7 +29,11 @@ if(CONFIG_NCS_IS_VARIANT_IMAGE)
   else()
     include(${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/bootloader_dts_utils.cmake)
 
-    dt_chosen(code_partition PROPERTY "zephyr,code-partition")
+    dt_chosen(code_partition PROPERTY "fw-to-relocate")
+    if("${code_partition}" STREQUAL "")
+      dt_chosen(code_partition PROPERTY "zephyr,code-partition")
+    endif()
+
```

Comment:
why is all of this done under variant image ?

All the adjustments in this `kconfig.cmake` module was made so that one highlevel build can produce one image, and from there also create a variant image which uses the exact same configuration.

I don't see how `fw-to-relocated` fits into this ?

Reply:
To my understanding (though I may be wrong), in the TCM-loader-enabled mode we have:
 - zephyr,code-partition: pointing to the same RAM memory, for both slots
  - fw-to-relocate: pointing to the MRAM, where the binary to be copied is stored
  
  Since this logic is used when there are variant apps built, it means that we are in the Direct XIP update mode(s).
  
  So now - as the zephyr,code-partition must be different for slot 0 and slot 1 in a regular Direct XIP, the fw-to-relocate must be different (to copy from either slot 0 or slot 1) once TCM-loader-enabled mode is used.